src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -499,11 +498,13 @@ public TextReader GetTextReader()
 
                             if (t == typeof(string))
                             {
-                                bool needsQuotes = ((string)cell).IndexOf(separator) != -1 || ((string)cell).IndexOf('\n') != -1;
+                                string stringCell = (string)cell;
+                                bool needsQuotes = stringCell.IndexOf(separator) != -1 || stringCell.IndexOf('\n') != -1 || stringCell.IndexOf('\"') != -1;
```

Comment:
It may be good to test the perf, but this could also be written as

```C#
bool needsQuotes = stringCell.AsSpan().IndexOfAny(separator, `\n`, '\"') != -1;
```

Making use of https://github.com/dotnet/runtime/blob/7fc8fb56327e9696ce9baa80f997e3f50675af1f/src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs#L1257-L1290